src_c/time.c

```diff
@@ -140,24 +132,28 @@ time_get_ticks(PyObject *self)
 static PyObject *
 time_delay(PyObject *self, PyObject *arg)
 {
-    int ticks;
+    double ticks;
     PyObject *arg0;
 
     /*for some reason PyArg_ParseTuple is puking on -1's! BLARG!*/
     if (PyTuple_Size(arg) != 1)
-        return RAISE(PyExc_ValueError, "delay requires one integer argument");
-    arg0 = PyTuple_GET_ITEM(arg, 0);
-    if (!PyInt_Check(arg0))
-        return RAISE(PyExc_TypeError, "delay requires one integer argument");
+        return RAISE(PyExc_ValueError, "delay requires only one argument");
 
-    ticks = PyInt_AsLong(arg0);
-    if (ticks < 0)
-        ticks = 0;
+    arg0 = PyTuple_GET_ITEM(arg, 0);
+    if (PyInt_Check(arg0)) {
+        ticks = (double)PyInt_AsLong(arg0);
+        ticks = accurate_delay(ticks);
+        return PyInt_FromLong((long)ticks);
+    } 
+    else if (PyFloat_Check(arg0)) {
+        ticks = PyFloat_AsDouble(arg0);
+        ticks = accurate_delay(ticks);
+        return PyFloat_FromDouble(ticks);
+    } 
+    else {
+        return RAISE(PyExc_TypeError, "delay must be an integer or floating point value");
```

Comment:
Can it require a number instead? Probably using kwargs instead of this custom argument parsing.

Reply:
I will try using py argparse tuple

Reply:
Done 👍